packages/bee-q/src/components/dropdown/bq-dropdown.tsx

```diff
@@ -0,0 +1,164 @@
+import { h, Component, Element, Event, EventEmitter, Prop, Listen } from '@stencil/core';
+
+import { FloatingUIPlacement } from '../../services/interfaces';
+
+/**
+ * @part base - The component's internal wrapper.
+ * @part trigger - The `div` element used to display the trigger element
+ * @part panel - The `div` element used to display the panel element (bq-panel)
+ */
+@Component({
+  tag: 'bq-dropdown',
+  styleUrl: './scss/bq-dropdown.scss',
+  shadow: true,
+})
+export class BqDropdown {
+  // Own Properties
+  // ====================
+
+  private trigger: HTMLElement;
+  private panelElement: HTMLBqPanelElement;
+
+  // Reference to host HTML element
+  // ===================================
+
+  @Element() el!: HTMLBqDropdownElement;
+
+  // State() variables
+  // Inlined decorator, alphabetical order
+  // =======================================
+
+  // Public Property API
+  // ========================
+
+  /** Distance (px) between the panel and the trigger element. */
+  @Prop({ reflect: true }) panelDistance?: number = 4;
+
+  /** Position of the panel */
+  @Prop({ reflect: true }) panelPlacement?: FloatingUIPlacement = 'bottom-start';
+
+  /**
+   * If true, panel is visible.
+   * You can toggle this attribute to show/hide the panel.
+   */
+  @Prop({ reflect: true }) panelOpen?: boolean = false;
+
+  /**
+   * Determines whether the scrollbar is visible or hidden within the panel.
+   */
+  @Prop({ reflect: true }) panelScrollbar?: boolean = false;
+
+  // Prop lifecycle events
+  // =======================
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  /**
+   * Handler to be called to check if the `bq-panel` switches state (visible/hidden).
+   * @returns CustomEvent - with value `{ opened: boolean }`
+   */
+  @Event() bqPanelChange: EventEmitter<{ opened: boolean }>;
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  componentDidLoad() {
+    this.initFloatingUIFromPanel();
+  }
+
+  // Listeners
+  // ==============
+
+  /** On click outside the panel */
+  @Listen('click', { target: 'document', passive: true })
+  onClickOutsidePanel(event: MouseEvent) {
+    // close the panel on click, except click within the panel or within the trigger element
+    if (!event.composedPath().includes(this.panelElement) && !event.composedPath().includes(this.trigger)) {
+      this.panelElement.open = false;
+    }
+  }
```

Comment:
Since `bq-panel` receives an `open` property that internally handles when to show/hide the panel content, we can just do something like this:

```suggestion
  /** Listens for the 'click' event on the document object
   * and closes the dropdown panel if the click is outside the component. */
  @Listen('click', { target: 'document', passive: true })
  onClickOutside(event: MouseEvent) {
    if (!this.open) return;
    // Close when clicking outside of the close element
    const path = event.composedPath();
    if (!path.includes(this.el)) {
      this.open = false;
    }
  }
```